e2e/nomostest/nt.go

```diff
@@ -425,25 +437,16 @@ func (nt *NT) MustMergePatch(obj client.Object, patch string, opts ...client.Pat
 	}
 }
 
-// DefaultRootSyncObjectCount returns the number of objects expected to be in
-// the default RootSync repo, based on how many RepoSyncs it's managing.
-func (nt *NT) DefaultRootSyncObjectCount() int {
-	numObjects := 2 // 2 for the 'safety' Namespace & 'safety' ClusteRole
-
-	// Count the number of unique RepoSyncs
-	numRepoSyncs := len(nt.NonRootRepos)
-
-	// Account for objects added to the default RootRepo by setupCentralizedControl
-	if nt.Control == ntopts.CentralControl && numRepoSyncs > 0 {
-		numObjects++                             // 1 for the ClusterRole, shared by all RepoSyncs
-		numObjects += nt.NumRepoSyncNamespaces() // 1 for each unique RepoSync Namespace
-		numObjects += numRepoSyncs               // 1 for each RepoSync
-		numObjects += numRepoSyncs               // 1 for each RepoSync RoleBinding
-		if isPSPCluster() {
-			numObjects += numRepoSyncs // 1 for each RepoSync ClusterRoleBinding
-		}
-	}
-	return numObjects
+// DefaultRootSyncObjectCount returns the number of declared resource objects in
+// the specified RootSync, according to records from test setup.
+func (nt *NT) DefaultRootSyncObjectCount(syncName string) int {
```

Comment:
I think something like `InitialR*SyncObjectCount` might be a bit more clear than `DefaultR*SyncObjectCount`

Reply:
I had a brain wave so I'm refactoring these to `nt.expectedObjects` and `ExpectedRootSyncObjectCount` and adding Add & Remove methods.

Reply:
ack, let me know when it's ready for review